core/Archive.php

```diff
@@ -449,6 +449,10 @@ public static function createDataTableFromArchive($recordName, $idSite, $period,
     {
         Piwik::checkUserHasViewAccess($idSite);
 
+        if (!empty($idSubtable) && !is_numeric($idSubtable)) {
+            throw new \Exception("idSubtable needs to be a number, '$idSubtable' given.");
```

Comment:
If I remember correctly there were cases were the idSubtable might not be numeric. Like the special case `all` to fetch all subtables: https://github.com/matomo-org/matomo/blob/680db1156be27769709bf9a6e7f67636e7cfb8dc/core/Archive.php#L115

Reply:
Good spot, I'd missed that. :+1: I've adjusted the check and the test to allow for this special case.